clients/python-client/python_client/kuberay_job_api.py

```diff
@@ -0,0 +1,196 @@
+"""
+Set of APIs to manage rayjobs.
+"""
+
+import logging
+import time
+from kubernetes import client, config
+from kubernetes.client.rest import ApiException
+from typing import Any, Optional
+from python_client import constants
+
+
+log = logging.getLogger(__name__)
+if logging.getLevelName(log.level) == "NOTSET":
+    logging.basicConfig(format="%(asctime)s %(message)s", level=constants.LOGLEVEL)
+
+TERMINAL_JOB_STATUSES = [
+    "STOPPED",
+    "SUCCEEDED",
+    "FAILED",
+]
+
+
+class RayjobApi:
+    """
+    RayjobApi provides APIs to list, get, create, build, update, delete rayjobs.
+    Methods:
+    - submit_job(entrypoint: str, ...) -> str: Submit and execute a job asynchronously.
+    - TODO: stop_job(job_id: str) -> (bool, str): Request a job to exit asynchronously.
+    - get_job_status(job_id: str) -> str: Get the most recent status of a job.
+    - wait_until_job_finished(job_id: str) -> bool: Wait until a job is completed.
+    - get_job_info(job_id: str): Get the latest status and other information associated with a job.
+    - TODO: list_jobs() -> List[JobDetails]: List all jobs along with their status and other information.
+    - TODO: get_job_logs(job_id: str) -> str: Get all logs produced by a job.
+    - TODO: tail_job_logs(job_id: str) -> Iterator[str]: Get an iterator that follows the logs of a job.
+    - delete_job(job_id: str) -> (bool, str): Delete a job in a terminal state and all of its associated data.
+    """
+
+    # initial config to setup the kube client
+    def __init__(self):
+        # loading the config
+        try:
+            self.kube_config: Optional[Any] = config.load_kube_config()
+        except config.ConfigException:
+            # No kubeconfig found, try in-cluster config
+            try:
+                self.kube_config: Optional[Any] = config.load_incluster_config()
+            except config.ConfigException:
+                log.error("Failed to load both kubeconfig and in-cluster config")
+                raise
+
+        self.api = client.CustomObjectsApi()
+
+    def __del__(self):
+        self.api = None
+        self.kube_config = None
+
+    def submit_job(self, k8s_namespace: str = "default", job: Any = None) -> Any:
+        """Submit a Ray job to a given namespace."""
```

Comment:
for consistency sake - can you add a bit more documentation about this function, same as for `get_job_status` function